src/java.base/share/classes/java/lang/Process.java

```diff
@@ -475,6 +476,35 @@ public boolean waitFor(long timeout, TimeUnit unit)
         return false;
     }
 
+    /**
+     * Causes the current thread to wait, if necessary, until the
+     * process represented by this {@code Process} object has
+     * terminated, or the specified waiting duration elapses.
+     *
+     * <p>If the process has already terminated then this method returns
+     * immediately with the value {@code true}.  If the process has not
+     * terminated and the duration is not positive, then
+     * this method returns immediately with the value {@code false}.
+     *
+     * <p>The default implementation of this method polls the {@code exitValue}
+     * to check if the process has terminated. Concrete implementations of this
+     * class are strongly encouraged to override this method with a more
+     * efficient implementation.
+     *
+     * @param duration the maximum duration to wait; if not positive,
+     *                this method returns immediately.
+     * @return {@code true} if the process has exited and {@code false} if
+     *         the waiting duration elapsed before the process has exited.
+     * @throws InterruptedException if the current thread is interrupted
+     *         while waiting.
+     * @throws NullPointerException if duration is null
+     * @since 24
+     */
+    public boolean waitFor(Duration duration) throws InterruptedException {
+        Objects.requireNonNull(duration, "duration");
+        return waitFor(TimeUnit.NANOSECONDS.convert(duration), TimeUnit.NANOSECONDS);
```

Comment:
With this change, the update to ProcessTools is unnecessary and could be backed out.

Reply:
The `ProcessTools` override is a delegating override; I think it still makes sense and should be kept.

Reply:
I think the test tool should not depend on the internal implementation, so I think it should be kept.

Reply:
Its still unnecessary, adds bulk and maintenance overhead that will go unused.

Reply:
I will leave it as is for now, as it is consistent with other `waitFor` overrides.